dashboard/src/actions/catalog.ts

```diff
@@ -60,16 +62,44 @@ const actions = [
 
 export type ServiceCatalogAction = ActionType<typeof actions[number]>;
 
+function isEmptyDeep(obj: any): boolean {
```

Comment:
I know that we are testing the overall behavior at the action level but could we add unit tests for this function?

Reply:
I don't know. There should be a way to unit test functions without exporting them, otherwise we are not unit testing.

Can't we make it work like this?
```
wrapper = shallow("foo")
instance = wrapper.instance();
instance.isEmptyDeep(...) ?
```

The thing is that this function is complex enough too deserve its unit test. Relying on functional/integration test might not be the best option once the complexity starts growing.

Reply:
moving this function to a different helper to be able to test it